spicy/zeek.spicy

```diff
@@ -43,37 +43,59 @@ public function flip_roles() : void &cxxname="spicy::zeek::rt::flip_roles";
 ## Returns the number of packets seen so far on the current side of the current connection.
 public function number_packets() : uint64 &cxxname="spicy::zeek::rt::number_packets";
 
+## Opaque handle to a protocol analyzer.
+public type ProtocolHandle = __library_type("spicy::zeek::rt::ProtocolHandle");
+
 ## Adds a Zeek-side child protocol analyzer to the current connection.
 ##
+## If the same analyzer was added previously with protocol_handle_get_or_create or
+## protocol_begin with same argument, and not closed with protocol_handle_close
+## or protocol_end, no new analyzer will be added.
+##
 ## analyzer: type of analyzer to instantiate, specified through its Zeek-side
 ## name (similar to what Zeek's signature action `enable` takes); if not
 ## specified, Zeek will perform its usual dynamic protocol detection to figure
 ## out how to parse the data (the latter will work only for TCP protocols, though.)
 public function protocol_begin(analyzer: optional<string> = Null) : void &cxxname="spicy::zeek::rt::protocol_begin";
```

Comment:
Document the error semantics here. I see in the code, that errors trigger exceptions.